### PR TITLE
Avatar Upload/Get from AWS S3

### PR DESCRIPTION
**Pull request by nicktran6997**
- [x] Setup dropzone on frontend (FE)
- [x] Validate that proper image is sent (FE)
- [x] Add avatar field to UserSchema (BE)
- [x] Add default avatar (FS)
- [x] Have app Avatar use image connected to User's avatar field (FS)
- [x] Rework GET/PUT request for getting profile-pic and updating profile pic respectively (add them to users route instead of upload route?) (BE)
- [x] Apply proper styling to Dialog. (FE)
- [x] Apply more specific image validation (only let one image get uploaded)
- [ ] Give option to unset profile picture to default.

--Other Changes Added that were Related to Profile Pictures--
- [x] Dropdown Menu appearing when clicking on profile picture
- [x] Logout Button attached to dropdown menu
- [x] Snackbar notification when picture has changed

Edit: Ok so generally everything works and looks fine as is. So I'll put up a proper PR for it now. The most I'll add is either merging with dev or fixing up some styling.

### CODE REVIEW

File: client/src/pages/layouts/components/AvatarDialogForm.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import {DropzoneDialog} from 'material-ui-dropzone';
+import axios from 'axios';
+
+const AvatarDialogForm = (props) => {
+  const {open, setAvatar, closeMenu, handleOpenSnackbar} = props;
+
+  const handleClose = () => {
+    // setOpen(false);
```
Comment (sina-jamshidi) [inline client/src/pages/layouts/components/AvatarDialogForm.js]: remove comment

File: client/src/pages/layouts/components/AvatarDialogForm.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    closeMenu();
+  };
+
+  const handleSave = (file) => {
+    const formData = new FormData();
+    formData.append('avatar', file[0]);
+    axios
+      .put('user/avatar', formData)
+      .then((resp) => setAvatar(resp.data.avatar))
+      .then(() => handleOpenSnackbar());
```
Comment (sina-jamshidi) [inline client/src/pages/layouts/components/AvatarDialogForm.js]: here maybe we also wanna call `closeMenu()`?

File: client/src/pages/layouts/components/ProfileAvatar.jsx [hunk 3]
```diff
@@ -1,9 +1,47 @@
...
+                  alt={value.user.name}
+                  src={avatar || value.user.avatar}
+                />
+              </Button>
+
+              {Boolean(anchorElem) && (
+                <UserMenu
+                  anchorElem={anchorElem}
+                  setAnchorElem={setAnchorElem}
+                  setAvatar={setAvatar}
```
Comment (sina-jamshidi) [inline client/src/pages/layouts/components/ProfileAvatar.jsx]: Nice!

File: server/routes/api/upload.js [hunk 4]
```diff
@@ -19,17 +16,29 @@ const upload = multer({
...
+// //@desc: get a profile picture from aws bucket
+// //@accesss private
+// router.get('/', (req, res, next) => {
+//   s3.getObject({Bucket: process.env.BUCKETNAME, Key: "download.png"}, (err, data) => {
+//     if (err) {
+//       return res.send({"error": err})
+//     }
+//     res.send({data})
+//   })
+// })
```
Comment (sina-jamshidi) [inline server/routes/api/upload.js]: remove comments

**Review by sina-jamshidi — APPROVED** [review]
Looks great! Looks like there's some overlap between this and the board UI. Maybe when one is merged we should make sure to check for conflicts with the other one

File: client/src/pages/layouts/components/UserMenu.jsx [hunk 5]
```diff
@@ -38,8 +51,38 @@ const UserMenu = (props) => {
...
+        action={
+          <>
+            <Button
+              color="secondary"
+              size="small"
+              onClick={handleCloseSnackbar}
+            >
+              UNDO
+            </Button>
+            <IconButton
```
Comment (swathivottigunta) [inline client/src/pages/layouts/components/UserMenu.jsx]: Hey Nick, Great Job!! Cool!!. I figured out two things
1. Once the profile photo submitted and updated, still upload dialog is open.
2. Can you clarify the purpose of the 'undo' button?

File: client/src/pages/layouts/components/ProfileAvatar.jsx [hunk 6]
```diff
@@ -1,9 +1,47 @@
...
+          }
+          return (
+            <div>
+              <Button onClick={handleClickOpen}>
+                <Avatar
+                  alt={value.user.name}
+                  src={avatar || value.user.avatar}
+                />
+              </Button>
+
```
Comment (swathivottigunta) [inline client/src/pages/layouts/components/ProfileAvatar.jsx]: Hey I uploaded the avatar photo successfully, and I see the avatar key entry in my user model as well. After I re-login, the profile picture is loaded with the default photo, not my avatar photo.

File: client/src/pages/layouts/components/UserMenu.jsx (hunk 5 above)
Comment (nicktran6997) [reply client/src/pages/layouts/components/UserMenu.jsx]: Hi Swathi! Thanks for the input :)  
For 1) I fixed the issue so the menu closes, and 2) It was basically just another version of closing the Snackbar which is kind of unnecessary so I removed it too. Thanks for pointing that out.

File: client/src/pages/layouts/components/ProfileAvatar.jsx (hunk 6 above)
Comment (nicktran6997) [reply client/src/pages/layouts/components/ProfileAvatar.jsx]: Just fixed the issue too, turns out the avatar field doesnt get passed on initial user-login, but handled that issue 👍

**Pull request merged**